### DISCUSSION TITLE
Ability to choose GPU device of osu!.exe from root folder.

### DISCUSSION THREAD

**Discussion by ChiNoel-osu** [ppy/osu#24825] [Ideas] [answered]
People who has a MUX switch (or MS Hybrid) enabled laptop can choose a GPU for their game. And for modern gaming laptops that have a good enough iGPU, in theory, running off iGPU might give a lower latency (in hybrid mode). Even if it don't, iGPU sips power compared to dGPU.
osu!lazer runs on dGPU by default on my system. And the actual game executable seems to be the one in the version folder like `.\app-2023.914.0\osu!.exe` instead of `.\osu!.exe` and after every update, settings to choose which GPU to use will lose its effect.
And no, choosing the GPU device for `.\osu!.exe` doesn't work.
I think I've seen an issue regarding this before but I can't seem to find it. But still, addressing this would be a huge convenience for some gamers.

**Comment by peppy** [comment]
This will be resolved when we upgrade squirrel to the latest release, as the structure has changed.

**Comment by ChiNoel-osu** [comment]
hmm...sorry to bump this discussion but we still need to manually select desired GPU for osu! from the version folder as of [2024.718.0](https://github.com/ppy/osu/releases/tag/2024.718.0).
If squirrel is a problem, I wonder if we can have a in-game workaround setting that sets this manually, as it can be altered through the registry.
`Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\DirectX\UserGpuPreferences`
![image](https://github.com/user-attachments/assets/9e4bc38c-6f7b-44a4-ba16-3bc11197aac3)
where on my system, 1 is Power Saving (iGPU), 2 is dGPU, and 0 is "let Windows decide".

**Comment by ChiNoel-osu** [comment, marked as answer]
Coming to close this discussion, Velopack rocks!
